src/components/Registration/Registration.jsx

```diff
@@ -0,0 +1,148 @@
+import React, { Component } from 'react';
+import './Registration.css';
+import { withRouter } from 'react-router-dom';
+import PropTypes from 'prop-types';
+import axios from 'axios';
+import { withAlert } from 'react-alert';
+import { compose } from 'redux';
+
+export class Registration extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      firstName: '',
+      lastName: '',
+      email: '',
+      password: '',
+      registerError: false,
+    };
+  }
+
+  handleInputChange = (e) => {
+    const { name, value } = e.target;
+    this.setState(() => ({ [name]: value }));
+  };
+
+  isValidEmail = (email) => {
+    const emailRegex = /^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$/,
+      doesEmailRegexMatchEmail = emailRegex.test(email);
+    return doesEmailRegexMatchEmail;
+  };
+
+  isValidPassword = (password) => {
+    const passwordRegex = /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)[a-zA-Z\d]{8,32}$/,
+      doesPasswordRegexMatchPassword = passwordRegex.test(password);
+    return doesPasswordRegexMatchPassword;
+  };
+
+  registerUser = async () => {
+    const {
+        firstName, lastName, email, password,
+      } = this.state,
+      requestBody = {
+        firstName,
+        lastName,
+        email,
+        password,
+      };
+
+    await axios.post('http://localhost:5005/api/v1.0/users', requestBody);
+  };
+
+  handleRegisterClick = () => {
+    if (this.isValidRegistration()) {
+      try {
+        this.registerUser();
+        this.setState({
+          firstName: '',
+          lastName: '',
+          email: '',
+          password: '',
+          registerError: false,
+        });
+        this.props.history.push('/login');
+      } catch (error) {
+        this.props.alert.show(error);
+      }
+    } else {
+      this.setState({ registerError: true });
+    }
+  };
+
+  isValidRegistration = () => {
+    const {
+      firstName, lastName, email, password,
+    } = this.state;
+    return firstName && lastName && this.isValidEmail(email) && this.isValidPassword(password);
+  };
+
+  render() {
+    const {
+      firstName, lastName, email, password, registerError,
+    } = this.state;
+    return (
+      <div className="register-page">
```

Comment:
The root className should be the name of the file. In this case 'Registeration'. This way if we ever need to wrap this class in a higher order component and need to target the css, it is easy to know the exact className to target instead of having to look into the file.